WorldBank-Transport/DRIVER
# Smart default calendar configs

## Overview
Fixes an issue where the calendar widget in Map and Records view would fail to work for languages that didn't have an explicit calendar configuration defined.

- Languages that do not have a calendar defined default to a basic config:
https://github.com/WorldBank-Transport/DRIVER/blob/d32b2b043da1d6f56c2254e7973599df247d2091/web/app/scripts/localization/date-utils.js#L9-L12
- Reduces room for configuration error by adding defaults for attributes of language configuration
  - Assumes that the calendar's language is the same as the language it's defined for (This, for example, helps prevent bugs like that the Nepalese calendar was configured for Lithuanian)
  - Assumes that the calendar is Gregorian unless otherwise specified

Because of the complexity of date format representations, this doesn't attempt to try to refactor that or introduce defaults (Like, say, that date segments are by default separated with `/` characters, or that dates are by default DMY).

### Checklist


### Demo
<img width="447" alt="screen shot 2019-02-26 at 12 36 36 pm" src="https://user-images.githubusercontent.com/1032849/53433831-33aea400-39c3-11e9-8a03-4c6a0e3df007.png">

### Notes
- Configures more languages for development environments to make foreign language testing easier
- e8dc05fc5814a13b5691866957ac5f71e3dd2621 alphabetizes the configuration list by language to make it easier to determine if a language is present or not. Unfortunately that means that the whole PR diff is very challenging to read, and may need to be looked at by commit individually

## Testing Instructions
- Log in to the UI
- Use the language dropdown to change language to "português do Brasil"
- Open Map or Record page
- Click on the "Intervalo de Datas" dropdown
  - It should open and default dates should be populated in the text boxes
- Click on a date text box
  - A date selection form should appear
- Select a date
  - The form should close and the text box should be updated with the new date

Closes [#164139074](https://www.pivotaltracker.com/story/show/164139074)
